subworkflows/local/prepare_genome.nf

```diff
@@ -93,17 +98,20 @@ workflow PREPARE_GENOME {
     if ('star' in prepare_tool_indices) {
         if (params.star_index) {
             if (params.star_index.endsWith('.tar.gz')) {
-                ch_star_index = UNTAR_STAR_INDEX ( params.star_index ).untar
+                UNTAR_STAR_INDEX (Channel.fromPath(params.star_index).map{ it -> [[id:it[0].baseName], it] })
+                ch_star_index = UNTAR_STAR_INDEX.out.untar.map{ meta, star_index -> [star_index] }.collect()
                 ch_versions   = ch_versions.mix(UNTAR_STAR_INDEX.out.versions)
             } else {
                 ch_star_index = file(params.star_index)
             }
         }
 
-        if((!ch_star_index) || getIndexVersion(ch_star_index) != '2.7.4a'){
-            ch_star_index   = STAR_GENOMEGENERATE(ch_fasta,ch_gtf).index
-            ch_versions     = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)
-        }
+        //print(getIndexVersion(ch_star_index))
+        //ch_star_index.view()
+        //if((!ch_star_index) || getIndexVersion(ch_star_index) != '2.7.4a'){
+        //    ch_star_index   = STAR_GENOMEGENERATE(ch_fasta,ch_gtf).index
+        //    ch_versions     = ch_versions.mix(STAR_GENOMEGENERATE.out.versions)
+        //}
```

Comment:
still no luck with this one?

Reply:
I think we can try to fix it in another PR, and merge this one so that we can go on with other things